src/utils/money.cljs

```diff
@@ -151,6 +152,11 @@
   [^js bn]
   (str "0x" (to-string bn 16)))
 
+(defn from-hex
+  [hex-str]
+  (-> (gint/fromString hex-str 16)
```

Comment:
We use `goog.math.Integer` for converting hex to decimal because JavaScript’s native number type has precision limitations for large integers (above 2^53). `goog.math.Integer` provides arbitrary-precision arithmetic, allowing accurate handling of large values without rounding errors, which is crucial for scenarios like cryptography or financial calculations.

Reply:
Maybe this belongs in the `utils.hex` ns

Reply:
@briansztamfater, can't we use the bignumber itself? The [documentation](https://mikemcl.github.io/bignumber.js/) says it can construct from hex string.

Also, it would be great to add a few tests to cover edge cases. Recently we experienced [exceptions](https://github.com/status-im/status-mobile/pull/21282) thrown by bignumber and it would be nice to avoid this.

Reply:
> Maybe this belongs in the `utils.hex` ns

@clauxx there are lots of hex-related functions on through different namespaces of the app. We need to define how we want to handle this. For now, as this is a function to create a BigNumber object based on a hex string, I put this on `utils.money` namespace, because that is the namespace where bignumber functions are. I feel we need to revisit all these namespaces, probably we need to move bignumber-related functions to a `utils.bignumber` namespace, but I didn't want to do this in this PR just to fix the bug. I will create an issue so we can do a proper refactor!

> @briansztamfater, can't we use the bignumber itself? The [documentation](https://mikemcl.github.io/bignumber.js/) says it can construct from hex string. excerpt from doc:
> 
> ```js
> new BigNumber('ff.8', 16)                  // '255.5'
> new BigNumber('0xff.8')                    // '255.5'
> ```
> 
> Also, it would be great to add a few tests to cover edge cases. Recently we experienced [exceptions](https://github.com/status-im/status-mobile/pull/21282) thrown by bignumber and it would be nice to avoid this.

@vkjr I didn't know we BigNumber had a constructor that admits hex strings. That's much cleaner indeed. I added some tests, and changed the approach to use the BigNumber constructor.